sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -649,6 +699,24 @@ def _raw_predict(self, X):
             dtype=self._baseline_prediction.dtype
         )
         raw_predictions += self._baseline_prediction
+
+        # bin categorical features when predicting outside of training loop
+        bin_categories = (not is_binned and
+                          self.categorical_features_ is not None)
+        if bin_categories:
+            X_binned_cat = self._bin_data(X, is_training_data=False,
```

Comment:
Why do we need to bin categorical features when predicting?

For the numerical features, what we store in the predictor node is the real-valued threshold (in contrast to the integer bin threshold), which avoids having to bin the data. Could a similar thing be done for categorical features, i.e. store the original categorical value instead of the encoded categorical value?